### ISSUE TITLE
Correctly pass ranges to Agda when operating over IPs

### ISSUE THREAD

**Issue by isovector** [isovector/cornelis#97]
The Agda interface needs a lot of love here:

* [ ] Mangle Ranges into the right representation for over the wire
* [ ] Attach buffer offsets to the ranges. I think we can get these via:

```viml
function! FileOffset()
    return line2byte(line('.')) + col('.') - 1
endfunction
```

* [ ] More correctly track IPs. These currently have an interval attached to them, but a better way would be to track them in extmarks.

**Comment by isovector** [comment]
To do the extmarks stuff, we'll need a map from interaction points to their extmarks. And then check whether `nvim_buf_get_extmark_by_id` with the `details` option correctly gives back the ending of its extmark after editing.

**Comment by isovector** [comment]
This might now be fixed by @Lysxia's new patch.

**Comment by Lysxia** [comment]
Right, the "attach buffer offsets" task seems to be subsumed by the new ability to convert between the two kinds of offsets at will.
